### PR TITLE
ci: Enable tests for iOS on GHA

### LINKED ISSUE #71617: iOS tests were dropped from CI during CircleCI->GHA migration
### 🐛 Describe the bug

I.e. this step
https://github.com/pytorch/pytorch/blob/9f0227a0ebde20eb50ff2ddc7e224a3a28a40ad2/.circleci/verbatim-sources/job-specs/job-specs-custom.yml#L523-L530
Could not be found anywhere in
https://github.com/pytorch/pytorch/blob/9f0227a0ebde20eb50ff2ddc7e224a3a28a40ad2/.github/templates/ios_ci_workflow.yml.j2#L84-L88

### Versions

N/A

cc @ezyang @gchanan @zou3519 @bdhirsh @seemethere @malfet @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71644

These were left out of the intial migration for some reason so this just
transfers over those tests

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Review by xta0 — APPROVED** [review]
LGTM

File: .github/templates/ios_ci_workflow.yml.j2 [hunk 1]
```diff
@@ -55,11 +57,33 @@ jobs:
...
+            *coreml*)
+              echo "USE_COREML_DELEGATE=1" >> "${GITHUB_ENV}"
+              ;;
+          esac
       - name: Install ios / conda Dependencies
         run: |
           # Install dependencies
           brew install libtool
           conda install numpy ninja pyyaml mkl mkl-include setuptools cmake cffi requests typing_extensions --yes
+          sudo gem install xcodeproj
```
Comment (seemethere) [inline .github/templates/ios_ci_workflow.yml.j2:86]: I guess `xcodeproj` was installed by default in CircleCI builds

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
> Merge failed due to Command '['git', '-C', '/home/runner/work/pytorch/pytorch', 'push', 'origin', 'master']' returned non-zero exit status 1.

We should probably just include a link to the failed workflow job in the comment as well for debugging purposes

**Comment by malfet** [comment]
> Merge failed due to Command '['git', '-C', '/home/runner/work/pytorch/pytorch', 'push', 'origin', 'master']' returned non-zero exit status 1.

Land race: https://github.com/pytorch/pytorch/runs/4971115468?check_suite_focus=true

**Pull request closed without merging**